src/components/Overview.tsx

```diff
@@ -1,57 +1,120 @@
+
+            const url = `/api/rest?${urlParams.toString()}`;
+            const response = await fetch(url, {
+                method: 'GET',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            });
+
+            if (!response.ok) {
+                setBooks([]);
+                throw new Error(`HTTP error! Status: ${response.status}`);
             }
-        };
 
-        fetchData();
-    }, [location]);
+            const content = await response.json();
+            const books = content._embedded?.buecher || [];
+            setBooks(books);
+
+            if (books.length === 0) {
+                console.warn('Keine Bücher gefunden.');
+            }
+        } catch (error) {
+            console.error('Fehler beim Laden der Buchdaten:', error);
+            setBooks([]);
+        }
+    };
+
+    useEffect(() => {
+        fetchBooks(); // Initialer API-Aufruf
+    }, []);
+
+    const handleSearch = () => {
+        // Wenn weder ein Suchbegriff noch eine Kategorie ausgewählt wurde
+        if (searchTerm.trim() === '' && category.trim() === '') {
+            fetchBooks(); // Zeige alle Bücher
+        } else {
+            fetchBooks(searchTerm, category); // Filtere basierend auf Eingabe
+        }
+    };
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === 'Enter') {
+            handleSearch(); // Starte die Suche auch bei Enter
+        }
+    };
 
     return (
-        <div className="content">
-            <h1>Books</h1>
-            <div className="flex flex-wrap">
-                {books.map((book) => (
-                    <BookCard
-                        key={book._links.self.href}
-                        className="flex-item"
-                        book={book}
-                    />
-                ))}
-                {user.token && <AddCard className="flex-item" />}
+        <div className="content p-4">
+            <h1 className="text-2xl font-bold mb-4">Bücher</h1>
+            <div className="flex items-center space-x-2 searchbar mb-4">
+                {/* Suchfeld */}
+                <Input
+                    className="flex-1 p-2 bg-gray-100 rounded border border-gray-300 focus:outline-none focus:ring-2 focus:ring-mainColor"
+                    type="text"
+                    value={searchTerm}
+                    onChange={(e) => setSearchTerm(e.target.value)}
+                    onKeyDown={handleKeyDown}
+                    placeholder="Bücher suchen..."
+                />
+                {/* Dropdown für Kategorie */}
+                <select
+                    className="p-2 bg-gray-100 rounded border border-gray-300 focus:outline-none focus:ring-2 focus:ring-mainColor"
+                    value={category}
+                    onChange={(e) => setCategory(e.target.value)}
+                >
+                    <option value="">Filter</option>
```

Comment:
The Default option should be named something like "Alle Buchkategorien" instead of "Filter"

Reply:
I will change it quickly